pkg/collector/corechecks/servicediscovery/module/envs_test.go

```diff
@@ -103,3 +106,49 @@ func memfile(name string, b []byte) (int, error) {
 
 	return fd, nil
 }
+
+// TestTargetEnvs it checks reading of target environment variables only from /proc/<pid>/environ.
```

Comment:
AFAICS it doesn't test we don't read other variables than the ones we want.

Reply:
I've changed test to check that unexpected variables not present in result.